### ISSUE TITLE
MAINT: gitignore and pydarshan work

### ISSUE THREAD

**Issue by tylerjereddy** [darshan-hpc/darshan#408]
The `pydarshan` subdirectory has its own `.gitignore`, which is useful, but when building out the C libs of `darshan-util` and so on in the same checkout, the tracking is a bit of a mess with object files we don't care about, and could easily be cleaned up without affecting the `.gitignore` settings of the other parent directories (actually, they don't have `.gitignore` anyway, but I'm mostly focused on `pydarshan` here):

`git status`

![image](https://user-images.githubusercontent.com/7903078/123828919-36857080-d8bf-11eb-9f47-6b72a178759d.png)

**Comment by tylerjereddy** [comment]
This situation is still a mess on `pydarshan-devel` branch, even after we rebase. In some ways, it is even worse with the new autotools stuff causing changes to tracked files like `configure` as well

**Comment by jakobluettgau** [comment]
Was there a particular reason not to include a .gitignore also at the repository root?

I think most of the object files and other temporary or compiled files can be covered with a couple of simple rules.

The configure file probably shouldn't be in pydarshan-devel (it is not in main).
For CI this might merely mean to call ./prepare.sh once and have autotools in the environment installed.

As a sidenote: I would still maintain a seperate .gitignore for the pydarshan directory. On the one hand because it keeps patterns simple and on the other hand because it allows for easier refactoring such as moving the pydarshan directory around.

**Comment by tylerjereddy** [comment]
Yeah, a carefully crafted `.gitignore` should fix most of what I see I think.

Maybe `configure` wasn't tracked, you are probably right, but it was polluting my `git status` at least.

**Comment by shanedsnyder** [comment]
Relatedly, it looks like `darshan-util/configure` is somehow tracked again on `pydarshan-devel`, maybe something weird happened as part of the rebase or some other merge. We will need to clear that stuff up at some point before merging back to `main`.
